src/js/jsx/sections/layers/LayerFace.jsx

```diff
@@ -76,15 +76,11 @@ define(function (require, exports, module) {
         // Given that the face hasn't changed and no selected ancestor has changed, this
         // component only needs to re-render when going from having a collapsed ancestor
         // (i.e., being hidden) to not having one (i.e., becoming newly visible).
-        var hadCollapsedAncestor = document.layers.hasCollapsedAncestor(this.props.layer);
-        if (!hadCollapsedAncestor) {
-            return false;
-        }
-
-        var nextDocument = nextProps.document,
+        var hadCollapsedAncestor = document.layers.hasCollapsedAncestor(this.props.layer),
+            nextDocument = nextProps.document,
```

Comment:
We should move the nextDocument assignment to the `//Deeper selection changes` and use it in the childOfSelection if block as well.